example/lib/scan/scan_bloc.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:async';
+import 'package:blemulator_example/adapter/ble_adapter.dart';
+import 'package:blemulator_example/scan/scan_result.dart';
+import 'package:blemulator_example/scan/scan_result_view_model.dart';
+import 'package:bloc/bloc.dart';
+import './bloc.dart';
+
+class ScanBloc extends Bloc<ScanEvent, ScanState> {
+  BleAdapter _bleAdapter;
+  StreamSubscription _scanResultsSubscription;
+  Map<String, ScanResult> _scanResults = {};
+
+  ScanBloc(this._bleAdapter);
+
+  @override
+  ScanState get initialState => ScanState.initial();
```

Comment:
`ScanState.initial()`? I assume it initialises all values to some safe defaults, but then why not just make those optional in normal constructor?